kernel/src/fs/configfs/tsm.rs

```diff
@@ -0,0 +1,206 @@
+}
+
+impl Report {
+    fn new(name: SysStr) -> Arc<Self> {
+        let mut builder = SysAttrSetBuilder::new();
+        builder.add(SysStr::from("inblob"), SysPerms::DEFAULT_RW_ATTR_PERMS);
+        builder.add(SysStr::from("outblob"), SysPerms::DEFAULT_RO_ATTR_PERMS);
+        builder.add(SysStr::from("provider"), SysPerms::DEFAULT_RO_ATTR_PERMS);
+        builder.add(SysStr::from("generation"), SysPerms::DEFAULT_RO_ATTR_PERMS);
+        let attrs = builder.build().expect("Failed to build attribute set");
+
+        Arc::new_cyclic(|weak_self| {
+            let fields = NormalNodeFields::new(name, attrs, weak_self.clone());
+
+            Report {
+                fields,
+                data: RwLock::new(TsmProvider::new()),
+            }
+        })
+    }
+}
+
+inherit_sys_leaf_node!(Report, fields, {
+    fn perms(&self) -> SysPerms {
+        SysPerms::DEFAULT_RW_PERMS
+    }
+
+    fn read_attr_at(&self, name: &str, offset: usize, writer: &mut VmWriter) -> Result<usize> {
+        match name {
+            "provider" => {
+                let mut printer = VmPrinter::new_skip(writer, offset);
+                write!(printer, "{}\n", self.data.read().provider)?;
+                return Ok(printer.bytes_written());
+            }
+
+            "outblob" => {
+                let mut data = self.data.write();
+
+                if data.inblob_len != TSM_INBLOB_MAX {
+                    return Err(Error::AttributeError);
+                }
+
+                if offset == 0 {
+                    data.outblob =
+                        Some(tdx_get_quote(&data.inblob).map_err(|_| Error::AttributeError)?);
+                }
+
+                let outblob = data.outblob.as_ref().ok_or(Error::AttributeError)?;
+
+                let slice = &outblob[offset..];
+                let write_len = writer
+                    .write_fallible(&mut slice.into())
+                    .map_err(|_| Error::AttributeError)?;
+                return Ok(write_len);
+            }
+
+            "generation" => {
+                let mut printer = VmPrinter::new_skip(writer, offset);
+                write!(printer, "{}\n", self.data.read().generation)?;
+                return Ok(printer.bytes_written());
+            }
+
+            _ => {}
+        }
+
+        Err(Error::AttributeError)
+    }
+
+    fn write_attr(&self, name: &str, reader: &mut VmReader) -> Result<usize> {
+        if name == "inblob" {
+            let mut data = self.data.write();
+
+            let mut writer = VmWriter::from(&mut data.inblob[..]);
+            let read_len = reader
+                .read_fallible(&mut writer)
+                .map_err(|_| Error::AttributeError)?;
+
+            data.inblob_len = read_len;
+            data.generation = data.generation.wrapping_add(1);
+
+            return Ok(read_len);
+        }
+
+        Err(Error::AttributeError)
+    }
+});
+
+static TSM_SUBSYSTEM: Once<Arc<Tsm>> = Once::new();
+
+pub fn init_tsm_subsystem() {
```

Comment:
Simplify the name to `init`. The parent module calls `tsm::init()`. So there won't be any ambiguity which subsystem is initialized by this function.